### PR TITLE
mgr/dashboard: Add RGW user and bucket management features

### PR DESCRIPTION
**Pull request by votdev**
bildschirmfoto vom 2018-04-11 15-08-24

fireshot capture 25 - ceph - http___localhost_4200_ _rgw_user

fireshot capture 18 - ceph - http___localhost_4200_ _rgw_user_add

fireshot capture 14 - ceph - http___localhost_4200_ _rgw_user_edit_test

bildschirmfoto vom 2018-04-11 15-08-58

bildschirmfoto vom 2018-04-11 15-09-17

bildschirmfoto vom 2018-04-11 15-09-39

fireshot capture 16 - ceph - http___localhost_4200_ _rgw_bucket

fireshot capture 17 - ceph - http___localhost_4200_ _rgw_bucket_add

bildschirmfoto vom 2018-04-11 15-10-55

### CODE REVIEW

**Comment by votdev** [comment]
jenkins retest this please

**Comment by rjfd** [comment]
QA tests results: http://pulpito.ceph.com/rdias-2018-04-26_15:40:43-rados:mgr-wip-rdias-testing-distro-basic-smithi/

There is one failure but I think it's unrelated with this PR.

**Comment by rjfd** [comment]
jenkins retest this please

**Comment by rjfd** [comment]
QA tests all successful: http://pulpito.ceph.com/rdias-2018-04-27_08:08:19-rados:mgr-wip-rdias-testing-distro-basic-mira/

**Comment by ricardoasmarques** [comment]
minor: What about being more verbose when we don't have subusers, capabilities, etc...

Something like `There are no subusers.`:

screenshot from 2018-04-27 11-26-56

Instead of `Empty`:

screenshot from 2018-04-27 11-27-14

**Comment by ricardoasmarques** [comment]
minor: On RBD form, we decided to disable the submission on "Enter key pressed".

If we want consistency, we should disable it here too.

**Comment by ricardoasmarques** [comment]
minor: Shouldn't we use a `default` button to close the dialog, instead of a `primary` button?

IMO, the `close` action doesn't deserve so highlight here.

screenshot from 2018-04-27 11-39-22

File: src/pybind/mgr/dashboard/frontend/src/app/shared/directives/copy2clipboard-button.directive.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
+import { Directive, ElementRef, HostListener, Input, OnInit, Renderer2 } from '@angular/core';
+
+import { ToastsManager } from 'ng2-toastr';
+
+@Directive({
+  selector: '[cdCopy2ClipboardButton]'
+})
+export class Copy2ClipboardButtonDirective implements OnInit {
```
Comment (ricardoasmarques) [inline src/pybind/mgr/dashboard/frontend/src/app/shared/directives/copy2clipboard-button.directive.ts]: Adding a reusable component on shared module deserves a separate commit.

This helps the review, and also future maintenance.

(something to take into account in the future...)

**Comment by votdev** [comment]
@ricardoasmarques Your comments has been addressed. I will create a separate PR for the Copy2ClipboardButtonDirective directive.

**Comment by votdev** [comment]
@rjfd I'm now catching exceptions [more selective](https://github.com/ceph/ceph/pull/21351/commits/1e666f2b8893900fbfd4ff170300acdacfb710b9).

**Comment by votdev** [comment]
@ricardoasmarques A separate commit has been created for the [Copy2ClipboardButtonDirective](https://github.com/ceph/ceph/pull/21351/commits/ff9849a888a86241a95f0c034c52c6c627ff9483?diff=unified).

**Comment by LenzGr** [comment]
retest this please

**Pull request merged**